src/screens/Ticketing/Tickets/use-recent-tickets.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  PreassignedFareProduct,
+  UserProfile,
+} from '../../../reference-data/types';
+import {UserProfileWithCount} from '../Purchase/Travellers/use-user-count-state';
+import {useEffect, useMemo, useReducer} from 'react';
+import {
+  listRecentFareContracts,
+  RecentFareContract,
+} from '../../../api/fareContracts';
+import {TariffZoneWithMetadata} from '../Purchase/TariffZones';
+import {findReferenceDataById} from '../../../reference-data/utils';
+import {
+  RemoteConfigContextState,
+  useRemoteConfig,
+} from '../../../RemoteConfigContext';
+
+export type RecentTicket = {
+  preassignedFareProduct: PreassignedFareProduct;
+  fromTariffZone: TariffZoneWithMetadata;
+  toTariffZone: TariffZoneWithMetadata;
+  userProfilesWithCount: UserProfileWithCount[];
+};
+
+type State = {
+  error: boolean;
+  loading: boolean;
+  recentFareContracts: RecentFareContract[];
+};
+
+const initialState: State = {
+  error: false,
+  loading: true,
+  recentFareContracts: [],
+};
+
+type Action =
+  | {type: 'FETCH'}
+  | {type: 'ERROR'}
+  | {type: 'SUCCESS'; data: RecentFareContract[]};
+
+type Reducer = (prevState: State, action: Action) => State;
+
+const reducer: Reducer = (prevState, action): State => {
+  switch (action.type) {
+    case 'FETCH':
+      return {
+        ...prevState,
+        loading: true,
+        error: false,
+      };
+    case 'ERROR':
+      return {
+        ...prevState,
+        loading: false,
+        error: true,
+      };
+    case 'SUCCESS':
+      return {
+        ...prevState,
+        loading: false,
+        error: false,
+        recentFareContracts: action.data,
+      };
+  }
+};
+
+const mapUsers = (
+  users: {[user_profile: string]: string},
+  userProfiles: UserProfile[],
+) =>
+  Object.entries(users)
+    .map(([profileId, count]) => ({
+      ...findReferenceDataById(userProfiles, profileId),
+      count: parseInt(count),
+    }))
+    .filter((u): u is UserProfileWithCount => 'id' in u)
```

Comment:
Is this just for casting? Could it be done by explicitly setting generic in `map`? Or is it also a filter, and could potentially be reduce instead of map + filter. Might be micro-optimization though.

Reply:
It is not just casting. If the ticket for some reason references a product/tariff zone/user profile that is not in our reference data, then the response from `findReferenceDataById` will be undefined. This `filter` is for filtering out these tickets, and also get the correct typing after the filter.

With that said `reduce` may just as well be used, and could probably make it more readable? My last team had these `filter`-methods with type guards several places, so I may be a little affected by that 😅